workers/go/throughputstress/workflow.go

```diff
@@ -194,24 +201,29 @@ func ThroughputStressWorkflow(ctx workflow.Context, params *throughputstress.Wor
 				}
 				return nil
 			},
-		)
-		if err != nil {
+		); err != nil {
+			return nil, err
+		}
+
+		// Wait for all handlers to finish
+		if err := workflow.Await(ctx, func() bool {
+			return workflow.AllHandlersFinished(ctx)
+		}); err != nil {
 			return nil, err
 		}
 
 		// Possibly continue as new
 		if params.ContinueAsNewAfterEventCount > 0 && workflow.GetInfo(ctx).GetCurrentHistoryLength() >= params.ContinueAsNewAfterEventCount {
 			if i == 0 {
-				return nil, fmt.Errorf("trying to continue as new on first iteration, workflow will never end")
+				// If this is the first iteration, don't ContinueAsNew since the workflow would never complete.
+				workflow.GetLogger(ctx).Info("skipping ContinueAsNew since 1st iteration - will ContinueAsNew next iteration")
+				continue
 			}
 			params.InitialIteration = i
 			params.TimesContinued++
 			return nil, workflow.NewContinueAsNewError(ctx, "throughputStress", params)
 		}
 	}
-	// Expect the self signal local activity to be run.
-	signchan := workflow.GetSignalChannel(ctx, ASignal)
```

Comment:
There were warnings about unhandled Signals. So I added a signal handler instead. Let me know if this makes sense; I haven't worked with those much before.